src/colorizer/utils/react_utils.ts

```diff
@@ -199,34 +199,50 @@ export function useScrollShadow(shadowColor: string = "#00000030"): {
 const RECENT_COLLECTIONS_STORAGE_KEY = "recentDatasets";
 const MAX_RECENT_COLLECTIONS = 10;
 
-export type RecentCollection = {
+export type StoredRecentCollection = {
   /** The absolute URL path of the collection resource. */
   url: string;
   /**
    * The user input for the collection resource.
-   * If `null`, uses the existing label (if already in recent datasets) or reuse the URL (if new).
+   * If `undefined`, uses the existing label (if already in recent datasets) or reuses the URL (if new).
    */
   label: string;
 };
 
+export type RecentCollection = Partial<StoredRecentCollection> & {
+  url: string;
+};
```

Comment:
This looks like it is adding complexity to the code, for a very simple data structure.  I think the verbose explanatory comment you put in the PR is really needed in the code too.  From the names RecentCollection and StoredRecentCollection it's hard to tell why they differ by this one url property.  Or else just simplify the type to have all 3 fields and use class methods to keep it internally consistent?

Reply:
Naively, it seems like all you need is a function like `ensureLabelAndUrlConsistent`

Reply:
I ended up just removing the duplicated datastructure and just allowing the label to be undefined. This fortunately did not cause as many problems as I thought it would.